test/jdk/javax/swing/JFileChooser/FileSizeCheck.java

```diff
@@ -21,95 +21,232 @@
  * questions.
  */
 
-/*
- * @test
- * @bug 8288882
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
- * @requires (os.family == "linux")
- * @summary To test if the 1-Empty-File size shows 0.0 KB and other files show correct size.
- * @run main/manual FileSizeCheck
- */
-
+import java.awt.AWTException;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Robot;
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.lang.reflect.InvocationTargetException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
 
+import javax.swing.AbstractButton;
 import javax.swing.JFileChooser;
 import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JTable;
+import javax.swing.JToggleButton;
 import javax.swing.SwingUtilities;
 import javax.swing.WindowConstants;
 
+/*
+ * @test
+ * @bug 8288882
+ * @key headful
+ * @requires (os.family == "linux")
+ * @summary Verifies if the size of an empty file is shown as 0.0 KB
+ *          as well as checks the displayed file sizes are rounded up
+ * @run main FileSizeCheck
+ */
 public class FileSizeCheck {
-    private static Path[] tempFilePaths;
-    private static final String INSTRUCTIONS =
-            "Click on the \"Details\" button in right-top corner.\n\n" +
-                    "Scroll Down if required. \n\n" +
-                    "Test 1: If the size of 1st-Empty-File shows 0.0 KB\n" +
-                    "Test 2: If the size of 2nd-File-1-Byte shows 0.1 KB\n" +
-                    "Test 3: If the size of 3rd-File-160-Byte shows 0.2 KB\n" +
-                    "Test 3: If the size of 4th-File-299-Byte shows 0.3 KB\n" +
-                    "Test 4: If the size of 5th-File-900-Byte shows 0.9 KB\n" +
-                    "Test 6: If the size of 6th-File-901-Byte shows 1.0 KB\n" +
-                    "Test 7: If the size of 7th-File-999-KB shows 999.0 KB\n" +
-                    "Test 8: If the size of 8th-File-1000-KB shows 1.0 MB\n" +
-                    "Test 9: If the size of 9th-File-2.8-MB shows 2.8 MB\n\n" +
-                           "press PASS.\n\n";
-
-    public static void test() {
-        JFrame frame = new JFrame("JFileChooser File Size test");
-        JFileChooser fc = new JFileChooser();
-        fc.setControlButtonsAreShown(false);
-        Path dir = Paths.get(".");
-        String[] tempFilesName = {"1st-Empty-File", "2nd-File-1-Byte", "3rd-File-160-Byte",
-                "4th-File-299-Byte", "5th-File-900-Byte", "6th-File-901-Byte",
-                "7th-File-999-KB", "8th-File-1000-KB", "9th-File-2.8-MB"};
+    private enum FileSize {
+        F0(    0, "0.0 KB"),
```

Comment:
The text probably depends from the Locale? Do we need to set it explicitly?

Reply:
I does, indeed. I set the locale to `Locale.US`.